packages/flutter/test/gestures/serial_tap_test.dart

```diff
@@ -0,0 +1,320 @@
+  );
+
+  // Down/up pair 4: normal tap sequence far away from pair 1
+  const PointerDownEvent down4 = PointerDownEvent(
+    pointer: 4,
+    position: Offset(130.0, 130.0),
+  );
+
+  const PointerUpEvent up4 = PointerUpEvent(
+    pointer: 4,
+    position: Offset(131.0, 129.0),
+  );
+
+  // Down/move/up sequence 5: intervening motion
+  const PointerDownEvent down5 = PointerDownEvent(
+    pointer: 5,
+    position: Offset(10.0, 10.0),
+  );
+
+  const PointerMoveEvent move5 = PointerMoveEvent(
+    pointer: 5,
+    position: Offset(25.0, 25.0),
+  );
+
+  const PointerUpEvent up5 = PointerUpEvent(
+    pointer: 5,
+    position: Offset(25.0, 25.0),
+  );
+
+  // Down/up pair 7: normal tap sequence close to pair 1 but on secondary button
+  const PointerDownEvent down6 = PointerDownEvent(
+    pointer: 6,
+    position: Offset(10.0, 10.0),
+    buttons: kSecondaryMouseButton,
+  );
+
+  const PointerUpEvent up6 = PointerUpEvent(
+    pointer: 6,
+    position: Offset(11.0, 9.0),
+  );
+
+  testGesture('Recognizes serial taps', (GestureTester tester) {
+    serial.addPointer(down1);
+    tester.closeArena(1);
+    tester.route(down1);
+    tester.route(up1);
+    GestureBinding.instance!.gestureArena.sweep(1);
+    expect(events, <String>['down#1', 'up#1']);
+
+    events.clear();
+    tester.async.elapse(const Duration(milliseconds: 100));
+    serial.addPointer(down2);
+    tester.closeArena(2);
+    tester.route(down2);
+    tester.route(up2);
+    GestureBinding.instance!.gestureArena.sweep(2);
+    expect(events, <String>['down#2', 'up#2']);
+
+    events.clear();
+    tester.async.elapse(const Duration(milliseconds: 100));
+    serial.addPointer(down3);
+    tester.closeArena(3);
+    tester.route(down3);
+    tester.route(up3);
+    GestureBinding.instance!.gestureArena.sweep(3);
+    expect(events, <String>['down#3', 'up#3']);
+  });
+
+  testGesture('Does not steal tap gesture below it in the tree', (GestureTester tester) {
+    bool recognizedSingleTap = false;
+    final TapGestureRecognizer singleTap = TapGestureRecognizer()
+      ..onTap = () {
+        recognizedSingleTap = true;
+      };
+
+    singleTap.addPointer(down1);
+    serial.addPointer(down1);
+    tester.closeArena(1);
+    tester.route(down1);
+    tester.route(up1);
+    GestureBinding.instance!.gestureArena.sweep(1);
+    expect(events, <String>['down#1', 'up#1']);
+    expect(recognizedSingleTap, isTrue);
+  });
+
+  testGesture('Fires cancel if competing recognizer declares victory', (GestureTester tester) {
+    final WinningGestureRecognizer winner = WinningGestureRecognizer();
+    winner.addPointer(down1);
+    serial.addPointer(down1);
+    tester.closeArena(1);
+    tester.route(down1);
+    tester.route(up1);
+    GestureBinding.instance!.gestureArena.sweep(1);
+    expect(events, <String>['down#1', 'cancel#1']);
+  });
+
+  testGesture('Fires cancel on second tap and resets if competing with double-tap recognizer', (GestureTester tester) {
```

Comment:
I believe it only fires cancel if the DTR enters the arena first - not for every competition with a DTR.

Reply:
Fixed